src/main/java/io/cryostat/net/web/http/api/v2/TargetsPostHandler.java

```diff
@@ -171,13 +187,27 @@ public IntermediateResponse<ServiceRef> handle(RequestParameters params) throws
             cryostatAnnotations.put(AnnotationKey.REALM, CustomTargetPlatformClient.REALM);
             serviceRef.setCryostatAnnotations(cryostatAnnotations);
 
-            boolean v = customTargetPlatformClient.addTarget(serviceRef);
-            if (!v) {
-                throw new ApiException(400, "Duplicate connectUrl");
+            if (!dryRun) {
+                boolean v = customTargetPlatformClient.addTarget(serviceRef);
+                if (!v) {
+                    throw new ApiException(400, "Duplicate connectUrl");
+                }
             }
             return new IntermediateResponse<ServiceRef>().body(serviceRef);
         } catch (JvmIdGetException e) {
-            throw new ApiException(404, "Couldn't connect to target: " + e.getTarget());
+            if (AbstractAuthenticatedRequestHandler.isJmxAuthFailure(e)) {
+                throw new ApiException(420, "JMX Credentials Not Provided or Not Valid", e);
```

Comment:
This sounds like it's the same or very similar to what `427` is already used for?

Reply:
Oh yess it is actually same as 427 but if we return 427, on the front-end, the auth modal will pop up. I think this should not happen as the credentials are just for testing. So I chose another one like 420. What do you think?

Reply:
Hmm. That's a little unfortunate. Is there any reasonable way to keep the backend emitting a `427` but refactor the frontend so that that status code can be caught and ignored, or handled with a different error handler which doesn't cause the modal to appear?

Reply:
I think there could be a way. Maybe on the front-end, we could check the `dryrun` query-parameter is available and set to true? I could try to test it out with the `url` field of the returned response.

Reply:
That's possible, though it means that `Api.service` that handles the `427` is going to become a bit tightly coupled to this specific notion of a `dryrun` query on a request, which probably only happens in this one place. Better if an alternate error handler function can get passed from the target creation modal into the Api.service when the request is being made so that the knowledge of the dryrun parameter is kept encapsulated within the component that makes those specific requests with that specific query param.

Reply:
Right, I added some new changes for front-end error handler here: https://github.com/cryostatio/cryostat-web/pull/700